pkg/operators/filter/filter_test.go

```diff
@@ -0,0 +1,475 @@
+			name:      "==",
+			filter:    "abc==def",
+			dsName:    "",
+			fieldName: "abc",
+			op:        comparisonTypeMatch,
+			negate:    false,
+			value:     "def",
+		},
+		{
+			name:      "!=",
+			filter:    "abc!=def",
+			dsName:    "",
+			fieldName: "abc",
+			op:        comparisonTypeMatch,
+			negate:    true,
+			value:     "def",
+		},
+		{
+			name:      "<=",
+			filter:    "abc<=def",
+			dsName:    "",
+			fieldName: "abc",
+			op:        comparisonTypeLte,
+			negate:    false,
+			value:     "def",
+		},
+		{
+			name:      "<",
+			filter:    "abc<def",
+			dsName:    "",
+			fieldName: "abc",
+			op:        comparisonTypeLt,
+			negate:    false,
+			value:     "def",
+		},
+		{
+			name:      ">=",
+			filter:    "abc>=def",
+			dsName:    "",
+			fieldName: "abc",
+			op:        comparisonTypeGte,
+			negate:    false,
+			value:     "def",
+		},
+		{
+			name:      ">",
+			filter:    "abc>def",
+			dsName:    "",
+			fieldName: "abc",
+			op:        comparisonTypeGt,
+			negate:    false,
+			value:     "def",
+		},
+		{
+			name:      "~",
+			filter:    "abc~def",
+			dsName:    "",
+			fieldName: "abc",
+			op:        comparisonTypeRegex,
+			negate:    false,
+			value:     "def",
+		},
+		{
+			name:      "!~",
+			filter:    "abc~def",
+			dsName:    "",
+			fieldName: "abc",
+			op:        comparisonTypeRegex,
+			negate:    false,
+			value:     "def",
+		},
+	}
+
+	var dsName string
+	var fieldName string
+	var op comparisonType
+	var negate bool
+	var value string
+	var err error
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			dsName, fieldName, op, negate, value, err = extractFilter(tc.filter)
+			if tc.error {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tc.dsName, dsName)
+			assert.Equal(t, tc.fieldName, fieldName)
+			assert.Equal(t, tc.op, op)
+			assert.Equal(t, tc.negate, negate)
+			assert.Equal(t, tc.value, value)
+		})
+	}
+}
+
+func TestFilter(t *testing.T) {
+	testCaseData := struct {
+		stringValue  string
+		int64Value   int64
+		float64Value float64
+		boolValue    bool
+	}{
+		stringValue:  "abc",
+		int64Value:   123,
+		float64Value: 456.0,
+		boolValue:    true,
+	}
+	type testCase struct {
+		name         string
+		filterString string
+		match        bool
+		error        bool
+	}
+	testCases := []testCase{
+		{
+			name:         "incomplete filter",
+			filterString: "abc",
+			error:        true,
+		},
+		{
+			name:         "string match positive",
+			filterString: "stringValue==abc",
+			match:        true,
+		},
+		{
+			name:         "string match negative",
+			filterString: "stringValue==def",
+			match:        false,
+		},
+		{
+			name:         "string not match positive",
+			filterString: "stringValue!=def",
+			match:        true,
+		},
+		{
+			name:         "string not match negative",
+			filterString: "stringValue!=abc",
+			match:        false,
+		},
+		{
+			name:         "string lte positive",
```

Comment:
I'd rather keep it and not restrict it artificially - do you think chances of using it by accident are really that big? There are valid use cases IMHO (although it could also be done with the (much slower) regex feature).

Reply:
I actually don't know if it is useful. Currently no use cases comes into my mind. But I might miss something ofc

Reply:
Hmm, restrict the output to entries starting with certain characters (or a range) - like said, that could also be done using regex, but this is much faster.

Reply:
I don't have anything against it. It is just a part of my brain screams when it sees > < with strings :D